judge/views/contests.py

```diff
@@ -1157,3 +1162,61 @@ def post(self, request, *args, **kwargs):
             return HttpResponseRedirect(self.get_success_url())
         else:
             return self.render_to_response(self.get_context_data(object=self.object))
+
+
+class ExportContestSubmissions(ContestMixin, SingleObjectMixin, View):
+    def get_object(self, queryset=None):
+        contest = super().get_object(queryset)
+        if not contest.is_editable_by(self.request.user):
+            raise PermissionDenied()
+        return contest
+
+    def get(self, request, *args, **kwargs):
+        contest = self.get_object()
+        output_dir = self.export_contest_submissions(contest)
+        response = HttpResponse(File(open(output_dir, 'rb')),
+                                content_type='application/zip')
+        os.remove(output_dir)
+        response['Content-Disposition'] = 'attachment; filename="%s-submissions.zip"' % (contest.key)
+        return response
+
+    def export_contest_submissions(self, contest):
+        output_dir = tempfile.mktemp(suffix='.zip')
```

Comment:
More comments about the main logic of the function (which submission will be stored, how the folder will be structured, ...)